src/Workleap.OpenApi.MSBuild/SpectralManager.cs

```diff
@@ -0,0 +1,140 @@
+    {
+        Directory.CreateDirectory(Path.Combine(this._openApiToolsDirectoryPath, "spectral"));
+        Directory.CreateDirectory(this._toolDirectory);
+    }
+
+    private string GetSpectralFileName()
+    {
+        var osType = GetOperatingSystem();
+        var architecture = GetArchitecture();
+
+        if (osType == "linux")
+        {
+            var distro = File.Exists("/etc/os-release") ? File.ReadAllText("/etc/os-release") : string.Empty;
+            if (distro.Contains("Alpine Linux"))
+            {
+                osType = "alpine";
+                this._loggerWrapper.Helper.LogMessage(MessageImportance.Low, "Installing on Alpine Linux.");
+            }
+        }
+
+        var fileName = $"spectral-{osType}-{architecture}";
+
+        if (osType == "win")
+        {
+            fileName = "spectral.exe";
+            this._loggerWrapper.Helper.LogMessage(MessageImportance.Low, "Installing on Windows.");
+        }
+
+        return fileName;
+    }
+
+    private async Task DownloadFileAsync(string url, string destination, CancellationToken cancellationToken)
+    {
+        if (File.Exists(destination))
+        {
+            this._loggerWrapper.Helper.LogMessage(MessageImportance.Low, "File already exist");
+
+            return;
+        }
+
+        using var httpClient = new HttpClient();
+        using var response = await httpClient.GetAsync(url, cancellationToken);
+        response.EnsureSuccessStatusCode();
+
+        if (!response.IsSuccessStatusCode)
+        {
+            using var retryResponse = await httpClient.GetAsync(url, cancellationToken);
+
+            if (retryResponse.IsSuccessStatusCode)
+            {
+                await SaveFileFromResponseAsync(destination, retryResponse, cancellationToken);
+            }
+            else
+            {
+                throw new OpenApiTaskFailedException("Spectral could not be installed.");
+            }
+        }
+        else
+        {
+            await SaveFileFromResponseAsync(destination, response, cancellationToken);
+        }
+    }
+
+    private static async Task SaveFileFromResponseAsync(string destination, HttpResponseMessage response, CancellationToken cancellationToken)
+    {
+        using var fileTarget = new FileStream(destination, FileMode.Create, FileAccess.Write, FileShare.None);
+        using var fileStream = await response.Content.ReadAsStreamAsync();
+
+        await fileStream.CopyToAsync(fileTarget, 1024, cancellationToken);
+    }
+
+    private static string GetOperatingSystem()
+    {
+        if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux))
+        {
+            return "linux";
+        }
+
+        if (RuntimeInformation.IsOSPlatform(OSPlatform.OSX))
+        {
+            return "macos";
+        }
+
+        if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+        {
+            return "win";
+        }
+
+        return "unknown";
+    }
+
+    private static string GetArchitecture()
```

Comment:
Same thing here, we should return the existing `Architecture` enum value and throw if the CPU arch isn't supported